app/assets/javascripts/modules/users/components/signup-form.js

```diff
@@ -0,0 +1,52 @@
+import BaseComponent from '~/base/component';
+
+const PASSWORD_FIELD = '#user_password';
+const USERNAME_FIELD = '#user_username';
+const EMAIL_FIELD = '#user_email';
+const CONFIRMATION_PASSWORD_FIELD = '#user_password_confirmation';
+const SUBMIT_BUTTON = '#submit_btn';
+
+// UsersPasswordForm component that handles user password form
+// interactions.
+class UsersSignUpForm extends BaseComponent {
+  elements() {
+    this.$password = this.$el.find(PASSWORD_FIELD);
+    this.$passwordConfirmation = this.$el.find(CONFIRMATION_PASSWORD_FIELD);
+    this.$username = this.$el.find(USERNAME_FIELD);
+    this.$userEmail = this.$el.find(EMAIL_FIELD);
+    this.$submit = this.$el.find(SUBMIT_BUTTON);
+  }
+
+  events() {
+    this.$el.on('focusout', PASSWORD_FIELD, e => this.onfocusout(e));
+    this.$el.on('focusout', CONFIRMATION_PASSWORD_FIELD, e => this.onfocusout(e));
+    this.$el.on('focusout', USERNAME_FIELD, e => this.onfocusout(e));
+    this.$el.on('focusout', EMAIL_FIELD, e => this.onfocusout(e));
+  }
+
+  onfocusout() {
```

Comment:
I would suggest to rename it to `onFocusout`. The idea (at least that's what I do) is to start a function/method with `on` only if it's an event listener/callback. That will highlight that this is probably a a function that is used as an event listener/callback related to the `focusout` event. 

I know that in this case it wouldn't add much value because this component is very simple, but it's good to have consistency around.